src/models/mSeriesChallengeParticipant.ts

```diff
@@ -0,0 +1,110 @@
+import { ListMeta, Model, ModelList } from '../model'
+import { AuthenticatedResponse, SessionHandler } from '../session'
+
+export const enum MSeriesChallengeParticipantSorting {
+  ID = 'id',
+  Name = 'name',
+  JoinedAt = 'joinedAt'
+}
+
+export interface MSeriesChallengeParticipantData {
+  id: number
+  userId: number
+  mSeriesChallengeId: number
+  joinedAt: string
+  currentValue: number
+  rank: number | undefined
+  name: string
+}
+
+export interface MSeriesChallengeParticipantResponse extends AuthenticatedResponse {
+  mSeriesChallengeParticipant: MSeriesChallengeParticipantData
+}
+
+export interface MSeriesChallengeParticipantListResponse extends AuthenticatedResponse {
+  mSeriesChallengeParticipants: MSeriesChallengeParticipantData[]
+  mSeriesChallengeParticipantsMeta: MSeriesChallengeParticipantListResponseMeta
+}
+
+export interface MSeriesChallengeLeaderboardResponse extends AuthenticatedResponse {
+  mSeriesChallengeParticipants: MSeriesChallengeParticipantData[]
+  mSeriesChallengeParticipantsMeta: MSeriesChallengeLeaderboardResponseMeta
+}
+
+export interface MSeriesChallengeParticipantListResponseMeta extends ListMeta {
+  nameSearchQuery: string
+  sort: MSeriesChallengeParticipantSorting
+}
+
+export interface MSeriesChallengeLeaderboardResponseMeta {
+  ascending: boolean
+  limit: number
+  offset: number
+  totalCount: number
+}
+
+export class MSeriesChallengeParticipants extends ModelList<
+MSeriesChallengeParticipant,
+MSeriesChallengeParticipantData,
+MSeriesChallengeParticipantListResponseMeta
+> {
+  constructor (mSeriesChallengeParticipants: MSeriesChallengeParticipantData[], mSeriesChallengeParticipantsMeta: MSeriesChallengeParticipantListResponseMeta, sessionHandler: SessionHandler) {
+    super(MSeriesChallengeParticipant, mSeriesChallengeParticipants, mSeriesChallengeParticipantsMeta, sessionHandler)
+  }
+}
+
+export class MSeriesChallengeLeaderboardParticipants extends ModelList<
```

Comment:
Why does `MSeriesChallengeLeaderboardParticipants` exist when it's identical to `MSeriesChallengeParticipants`?

Reply:
`MSeriesChallengeLeaderboardParticipants` has different `meta` data (no `sort`)

Reply:
Okay